controllers/core/featureflagconfiguration/controller_test.go

```diff
@@ -0,0 +1,168 @@
+	}{
+		{
+			name: "no provider set + no owner set -> ffconfig and cm will be updated",
+			cm: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      cmName,
+					Namespace: testNamespace,
+					Annotations: map[string]string{
+						"openfeature.dev/featureflagconfiguration": ffConfigName,
+					},
+				},
+			},
+			ffConfig:     createTestFFConfig(ffConfigName, testNamespace, cmName, ""),
+			wantProvider: "flagd",
+			wantCM: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      cmName,
+					Namespace: testNamespace,
+					Annotations: map[string]string{
+						"openfeature.dev/featureflagconfiguration": ffConfigName,
+					},
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion: "core.openfeature.dev/v1alpha1",
+							Kind:       "FeatureFlagConfiguration",
+							Name:       ffConfigName,
+						},
+					},
+				},
+				Data: map[string]string{
+					v1alpha1.FeatureFlagConfigurationConfigMapKey(testNamespace, cmName): "spec",
+				},
+			},
+			cmDeleted: false,
+		},
+		{
+			name: "one owner ref set -> cm will be deleted",
+			cm: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      cmName,
+					Namespace: testNamespace,
+					Annotations: map[string]string{
+						"openfeature.dev/featureflagconfiguration": ffConfigName,
+					},
+				},
+			},
+			ffConfig:     createTestFFConfig(ffConfigName, testNamespace, cmName, ""),
+			wantProvider: "flagd",
+			wantCM: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      cmName,
+					Namespace: testNamespace,
+					Annotations: map[string]string{
+						"openfeature.dev/featureflagconfiguration": ffConfigName,
+					},
+				},
+			},
+			cmDeleted: true,
+		},
+	}
+
+	err := v1alpha1.AddToScheme(scheme.Scheme)
+	require.Nil(t, err)
+	req := ctrl.Request{
+		NamespacedName: types.NamespacedName{
+			Namespace: testNamespace,
+			Name:      ffConfigName,
+		},
+	}
+
+	ctx := context.TODO()
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakeClient := fake.NewClientBuilder().WithScheme(scheme.Scheme).WithObjects(tt.ffConfig, tt.cm).Build()
+
+			r := &FeatureFlagConfigurationReconciler{
+				Client: fakeClient,
+				Log:    ctrl.Log.WithName("featureflagconfiguration-controller"),
+				Scheme: fakeClient.Scheme(),
+			}
+
+			if tt.cmDeleted {
+				ffConfig2 := &v1alpha1.FeatureFlagConfiguration{}
+				err = fakeClient.Get(ctx, types.NamespacedName{Name: ffConfigName, Namespace: testNamespace}, ffConfig2)
+				require.Nil(t, err)
+
+				cm2 := &corev1.ConfigMap{}
+				err = fakeClient.Get(ctx, types.NamespacedName{Name: cmName, Namespace: testNamespace}, cm2)
+				require.Nil(t, err)
+
+				cm2.OwnerReferences = append(cm2.OwnerReferences, v1alpha1.GetFfReference(ffConfig2))
+				err := r.Client.Update(ctx, cm2)
+				require.Nil(t, err)
+			}
+
+			_, err = r.Reconcile(ctx, req)
+			require.Nil(t, err)
+
+			ffConfig2 := &v1alpha1.FeatureFlagConfiguration{}
```

Comment:
Is there more appropriate var naming than ffConfig2?

Reply:
fixed